File: angular/src/app/models/multa.ts
import { Agente } from './agente';
import { Vehiculo } from './vehiculo';

export class Multa {
    private _id: number;
    private _concepto: string;
    private _importe: number;
    private _fechaAlta: string;
    private _fechaModificacion: string;
    private _fechaBaja: string;
    private _agente: Agente;
    private _vehiculo: Vehiculo;

    constructor( 
        concepto: string, 
        importe: number, 
        fechaAlta: string, 
        agente: Agente,
        vehiculo: Vehiculo,
        id?: number,
        fechaModificacion?: string,
        fechaBaja?: string) {

        this._id = (id) ? id : -1;
        this._concepto = concepto;
        this._importe = importe;
        this._fechaAlta = fechaAlta;
        this._fechaModificacion = (fechaModificacion) ? fechaModificacion : undefined;
        this._fechaBaja = (fechaBaja) ? fechaBaja : undefined;
        this._agente = agente;
        this._vehiculo = vehiculo;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

    public get concepto(): string {
        return this._concepto;
    }
    public set concepto(value: string) {
        this._concepto = value;
    }

    public get importe(): number {
        return this._importe;
    }
    public set importe(value: number) {
        this._importe = value;
    }

    public get fechaAlta(): string {
        return this._fechaAlta;
    }
    public set fechaAlta(value: string) {
        this._fechaAlta = value;
    }

    public get fechaModificacion(): string {
        return this._fechaModificacion;
    }
    public set fechaModificacion(value: string) {
        this._fechaModificacion = value;
    }

    public get fechaBaja(): string {
        return this._fechaBaja;
    }
    public set fechaBaja(value: string) {
        this._fechaBaja = value;
    }

    public get agente(): Agente {
        return this._agente;
    }
    public set agente(value: Agente) {
        this._agente = value;
    }

    public get vehiculo(): Vehiculo {
        return this._vehiculo;
    }
    public set vehiculo(value: Vehiculo) {
        this._vehiculo = value;
    }
}

File: modelo/src/main/java/com/ipartek/formacion/modelo/pojo/Multa.java
package com.ipartek.formacion.modelo.pojo;

import java.util.Date;

import javax.validation.constraints.Digits;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

public class Multa {
	private int id;
	@NotNull
	@Digits(fraction = 2, integer = 5)
	private Double importe;
	@NotNull
	@Size(min=10, max=255)
	private String concepto;
	private Date fechaAlta;
	private Date fechaModificacion;
	private Date fechaBaja;
	private Agente agente;
	private Vehiculo vehiculo;
	
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public double getImporte() {
		return importe;
	}
	public void setImporte(Double importe) {
		this.importe = importe;
	}
	public String getConcepto() {
		return concepto;
	}
	public void setConcepto(String concepto) {
		this.concepto = concepto;
	}
	public Date getFechaAlta() {
		return fechaAlta;
	}
	public void setFechaAlta(Date fechaAlta) {
		this.fechaAlta = fechaAlta;
	}
	public Date getFechaModificacion() {
		return fechaModificacion;
	}
	public void setFechaModificacion(Date fechaModificacion) {
		this.fechaModificacion = fechaModificacion;
	}
	public Date getFechaBaja() {
		return fechaBaja;
	}
	public void setFechaBaja(Date fechaBaja) {
		this.fechaBaja = fechaBaja;
	}
	public Agente getAgente() {
		return agente;
	}
	public void setAgente(Agente agente) {
		this.agente = agente;
	}
	public Vehiculo getVehiculo() {
		return vehiculo;
	}
	public void setVehiculo(Vehiculo vehiculo) {
		this.vehiculo = vehiculo;
	}
	
	public Multa(int id, Double importe, String concepto, Date fechaAlta, Date fechaModificacion, Date fechaBaja, Agente agente, Vehiculo vehiculo) {
		this();
		setId(id);
		setImporte(importe);
		setConcepto(concepto);
		setFechaAlta(fechaAlta);
		setFechaModificacion(fechaModificacion);
		setFechaBaja(fechaBaja);
		setAgente(agente);
		setVehiculo(vehiculo);
	}
	
	public Multa() {
		super();
		this.id=-1;
		this.importe=0.0;
		this.concepto="";
		this.fechaAlta=null;
		this.fechaModificacion=null; 
		this.fechaBaja=null; 
		this.agente=new Agente();
		this.vehiculo=new Vehiculo();
	}
	
	@Override
	public String toString() {
		return "Multa [id=" + id + ", importe=" + importe + ", concepto=" + concepto + ", fechaAlta=" + fechaAlta + ", agente="
				+ agente + ", vehiculo=" + vehiculo + "]";
	}
	
	
	
	


}
